pkg/alertmanager/alertmanager.go

```diff
@@ -300,12 +306,42 @@ func New(cfg *Config, reg *prometheus.Registry) (*Alertmanager, error) {
 		am.mux.Handle(a, http.NotFoundHandler())
 	}
 
+	// This route is an experimental Mimir extension to the receivers, API, so we put
+	// it under an additional prefix to avoid any confusion with upstream Alertmanager.
+	if cfg.GrafanaAlertmanagerCompatibility {
+		am.mux.Handle(
+			path.Join(am.cfg.ExternalURL.Path, "/experimental/api/v1/receivers"),
+			http.HandlerFunc(am.GetReceiversHandler))
+	}
+
```

Comment:
Open to suggestions for:
- What path to use, using the `/experimental` prefix seemed safest to avoid potential collisions with upstream routes.
- Whether this should be gated behind `-alertmanager.grafana-alertmanager-compatibility-enabled` or something else.

Reply:
Good question. Traditionally, all the per-tenant APIs are located under `/alertmanager`, because there's a catch-all route defined [here](https://github.com/grafana/mimir/blob/ee069d247c7c22848ee668027c432a97e120cd9f/pkg/api/api.go#L207):

I'm not sure what I prefer. I like not polluting the upstream API space, but I like the consistency that all `/alertmanager` (or whatever prefix is used) are per-tenant.

I'll try changing the API path, I think I prefer that.